### ISSUE TITLE
Clang 13 build error

### ISSUE THREAD

**Issue by trolley813** [o3de/o3de#5235]
**Describe the bug**
Clang 13 introduced new `-Wunused-but-set-parameter` and `-Wunused-but-set-variable` warning flags, which emit when a parameter or a variable is set but not used. O3DE has `-Werror`, so the build fails with these new warnings as errors (since there are few probably unused variables).

**Steps to reproduce**
1. Configure the CMake build to use Clang 13 (e.g. with `-DCMAKE_C_COMPILER=clang-13` and `DCMAKE_CXX_COMPILER=clang++-13`)
2. Attempt to build the engine
3. See build errors

**Expected behavior**
Engine should build successfully with Clang 13 (it actually builds when explicitly disable the warnings by adding  `-Wno-unused-but-set-parameter` and `-Wno-unused-but-set-variable` into `cmake/Platforms/Common/Clang/Configurations_clang.cmake`).

**Actual behavior**
Build fails with errors described above.

**Assets required**
(not required, only the base engine)

**Screenshots/Video**
not applicable here

**Found in Branch**
development

**Desktop/Device (please complete the following information):**
 - Device: PC
 - OS: Linux
 - Version: Fedora 35
 - CPU: AMD Ryzen 9-3950x
 - GPU: Nvidia RTX 2080-super
 - Memory: 64GB

**Additional context**
Basically there are 2 options to fix that: either to disable the warnings as described above (this probably should not break the build with earlier Clang version since we have `-Wno-unknown-warning-option`), or to fix all the places with unused variables (but it may be infeasible in some cases).
